DuggaSys/sectioned.js

```diff
@@ -484,6 +484,72 @@ function refreshGithubRepo(courseid, user)
   return dataCheck;
 }
 
+//Send new Github URL and course id to PHP-script which gets and saves the latest commit in the sqllite db
+function updateGithubRepo(githubURL, cid) {
+	//Used to return success(true) or error(false) to the calling function
+	regexURL = githubURL.replace(/.git$/, "");
+  var dataCheck;
+	$.ajax({
+		async: false,
+		url: "../DuggaSys/gitcommitService.php",
+		type: "POST",
+		data: {'githubURL':regexURL, 'cid':cid, 'action':'directInsert'},
+		success: function() { 
+			//Returns true if the data and JSON is correct
+			dataCheck = true;
+		},
+		error: function(data){
+			//Check FetchGithubRepo for the meaning of the error code.
+			switch(data.status){
+				case 422:
+					alert(data.responseJSON.message + "\nDid not create/update course");
+					break;
+				case 503:
+					alert(data.responseJSON.message + "\nDid not create/update course");
+					break;
+				default:
+					alert("Something went wrong...");
```

Comment:
These should probably return different error messages. [HTTP codes](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status)

Reply:
case 422 and 503 could also be handled like this:
```
 switch($type) 
 {
     case 1:
     case 2:
     case 3:
         print 'success';
     break;
     case 4:
         print 'success for type 4';
     break;
 }
```